test/e2e/tests/address-book.spec.js

```diff
@@ -120,4 +121,96 @@ describe('Address Book', function () {
       },
     );
   });
+  it('Edit entry in address book', async function () {
+    await withFixtures(
+      {
+        fixtures: 'address-entry',
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.clickElement('.identicon__address-wrapper');
+        await driver.clickElement({ text: 'Settings', tag: 'div' });
+        await driver.clickElement({ text: 'Contacts', tag: 'div' });
+        await driver.clickElement('[data-testid="recipient"]');
+
+        await driver.clickElement({ text: 'Edit', tag: 'button' });
+        const inputUsername = await driver.findElement('#nickname');
+        await inputUsername.clear();
+        await inputUsername.fill('Test Name Edit');
+
+        const inputAddress = await driver.findElement('#address');
+        await inputAddress.clear();
+        await inputAddress.fill('0x74cE91B75935D6Bedc27eE002DeFa566c5946f74');
+
+        await driver.clickElement('[data-testid="page-container-footer-next"]');
+
+        const recipientUsername = await driver.findElement({
+          text: 'Test Name Edit',
+          tag: 'div',
+        });
+        assert.equal(
+          await recipientUsername.getText(),
+          'Test Name Edit',
+          'Username is not edited correctly',
+        );
+
+        const recipientAddress = await driver.findElement(
+          '.send__select-recipient-wrapper__group-item__subtitle',
+        );
+        assert.equal(
+          await recipientAddress.getText(),
+          '0x74cE...6f74',
+          'Recipient address is not edited correctly',
+        );
+      },
+    );
+  });
+  it('Adds an entry to address book and deletes it from address book', async function () {
+    await withFixtures(
+      {
+        fixtures: 'address-entry',
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.clickElement('.identicon__address-wrapper');
+        await driver.clickElement({ text: 'Settings', tag: 'div' });
+        await driver.clickElement({ text: 'Contacts', tag: 'div' });
+
+        const contacts = await driver.findElements(
+          '.send__select-recipient-wrapper__group-item',
+        );
+        assert.equal(contacts.length, 1);
+        // adding new account
```

Comment:
Since we are using fixtures, I suggest to skip the part of adding a new contact, since it's already there. So we just test that the Delete function works, as the Add Contact function has already been tested on the first `it` block.

So you could remove from line `193` to line `206` and on line `208` look for `Test Name 1` instead of `Test Name 2`
```
        await driver.clickElement({ text: 'Test Name 1', tag: 'div' });
        await driver.clickElement({ text: 'Edit', tag: 'button' });
        await driver.clickElement({ text: 'Delete account', tag: 'a' });
        // it checks if account is deleted
        assert.equal(contacts.length, 1);
```
The `it` block could then be re-written something like `Deletes existing entry from address book`

Reply:
Done

Reply:
LGTM!